cql/src/main/java/org/apache/cassandra/stargate/transport/internal/ProxyInfo.java

```diff
@@ -13,4 +18,12 @@ public ProxyInfo(InetSocketAddress publicAddress, InetSocketAddress sourceAddres
     this.publicAddress = publicAddress;
     this.sourceAddress = sourceAddress;
   }
+
+  public Map<String, String> toHeaders() {
```

Comment:
Using `Map<String, Object>` here and in the predicate will allow this us to keep this as an `InetSocketAddress` which might require less intermediate objects to convert in downstream code.